tests/sources/test_notion.py

```diff
@@ -740,13 +759,37 @@ async def test_advanced_rules_validation(advanced_rules, expected_validation_res
     async with create_source(
         NotionDataSource, notion_secret_key="secret_key"
     ) as source:
-        with patch(
-            "connectors.sources.notion.async_iterate_paginated_api",
+        with patch.object(
+            NotionClient,
+            "async_iterate_paginated_api",
             return_value=AsyncIterator([DATABASE]),
         ), patch.object(
             NotionDataSource, "get_entities", return_value=[DATABASE, BLOCK]
         ):
             validation_result = await NotionAdvancedRulesValidator(source).validate(
                 advanced_rules
             )
-        assert validation_result == expected_validation_result
+            assert validation_result == expected_validation_result
+
+
+@pytest.mark.asyncio
+async def test_async_iterate_paginated_api():
```

Comment:
Can you add tests to ensure that 429's are retried, but other errors are not? Perhaps also add a test to make sure that the original `async_iterate_paginated_api` is not being called anywhere?

Reply:
Updated via commit id: [2a601f1](https://github.com/elastic/connectors/pull/2297/commits/2a601f1935fd829667106e11af1621eac2c54482)